flocker/ca/test/test_ca.py

```diff
@@ -67,8 +351,8 @@ def test_decoded_certificate_matches_private_key(self):
         be paired with.
         """
         nc = NodeCredential.initialize(self.path, self.ca)
-        priv = nc.keypair.keypair.original
-        pub = nc.certificate.getPublicKey().original
+        priv = nc.credential.keypair.keypair.original
+        pub = nc.credential.certificate.getPublicKey().original
```

Comment:
I'm not sure what "decoded" is meant to convey in this test method name/docstring.  Is it to do with the ASN.1 decoding that happens lower down?  If so, that seems like a sort of uninteresting implementation detail of the test.  I think what's actually important here is that the certificate and private key match each other (ie, make up a key pair).  How this gets proven is best left as implementation details inside the test method.